pkg/daemon/sync.go

```diff
@@ -41,67 +44,103 @@ type changeSet struct {
 }
 
 // Sync starts the synchronization of the cluster with git.
-func (d *Daemon) Sync(ctx context.Context, started time.Time, newRevision string, ratchet revisionRatchet) error {
-	// Make a read-only clone used for this sync
+func (d *Daemon) Sync(ctx context.Context, started time.Time, newRevision string, ratchet ratchet) error {
+	currentRevision, err := ratchet.CurrentRevision(ctx)
+	if err != nil {
+		return err
+	}
+
 	ctxt, cancel := context.WithTimeout(ctx, d.GitTimeout)
-	working, err := d.Repo.Export(ctxt, newRevision)
+	defer cancel()
+
+	cleanupClone := func(clone *git.Export) {
+		if err := clone.Clean(); err != nil {
+			d.Logger.Log("error", fmt.Sprintf("cannot clean clone: %s", err))
+		}
+	}
+
+	// Get last-synced resources for comparison
+	lastResources, err := ratchet.CurrentResources(ctx)
 	if err != nil {
 		return err
 	}
-	cancel()
-	defer func() {
-		if err := working.Clean(); err != nil {
-			d.Logger.Log("error", fmt.Sprintf("cannot clean sync clone: %s", err))
+	// Resource map is not initialized
+	if lastResources == nil {
+		if currentRevision == "" {
+			// Initial clone
+			lastResources = make(map[string]resource.Resource)
+		} else {
+			// Clone last-synced revision
+			lastSynced, err := d.Repo.Export(ctxt, currentRevision)
+			if err != nil {
+				return err
+			}
+			defer cleanupClone(lastSynced)
+
+			lastResourceStore, err := d.getManifestStore(lastSynced)
```

Comment:
When processing the current revision, there is a step which unlocks secrets. Is that also necessary here?

Reply:
Fixed. Thank you.